client/my-sites/checkout/composite-checkout/test/checkout-main.tsx

```diff
@@ -47,6 +49,7 @@ describe( 'CheckoutMain', () => {
 	let MyCheckout;
 
 	beforeEach( () => {
+		dispatch( CHECKOUT_STORE ).reset();
```

Comment:
Alternatively, if we only have to do this for tests, we could use `jest.resetModules()` and re-import the store before every test, as demonstrated here: https://stackoverflow.com/a/48990084/3936800

I wonder why this is necessary, though. We should be able to mock specific selectors or actions in the store per-test if necessary, no?

Reply:
> we could use jest.resetModules() and re-import the store before every test

🤔  Interesting... I wonder if that might cause challenges based on the other things being imported/mocked in these tests though. An explicit reset of the store feels a little more clear to me than needing to re-mock a ton of selectors.

> I wonder why this is necessary, though. We should be able to mock specific selectors or actions in the store per-test if necessary, no?

These tests are integration tests so they are designed to operate as close to "real behavior" as possible to cover a of functionality. If we mock parts of the store there's a risk that a regression could occur in the store itself without being caught by these tests.

In most cases when checkout is loaded, this store will be empty (users do not often visit checkout, fill out the forms, then leave checkout, then come back) so these tests are designed to start from that premise. That way we can test behavior like auto-fill without needing to clear specific data (or to put it another way, this is clearing that specific data).